File: Python/1456.py
# Maximum Number of Vowels in a Substring of Given Length
# Medium

def maxVowels(s: str, k: int) -> int:
    
    dt = {
        'a': 1,
        'e': 1,
        'i': 1,
        'o': 1,
        'u': 1,
    }

    ls = [dt.get(x, 0) for x in s]
    max_vowels = sum(ls[0:k])
    sum_vowels = max_vowels
    for i in range(1, len(ls) - k + 1):
        j = i + k - 1
        sum_vowels = sum_vowels - ls[i - 1] + ls[j]
        if max_vowels < sum_vowels:
            max_vowels = sum_vowels
    return max_vowels

File: C++/1456.cpp
// Maximum Number of Vowels in a Substring of Given Length
// Medium

#include <map>
#include <string>

using namespace std;

int maxVowels(string s, int k) {
    
    int arr[s.length()];

    int i = 0;
    for ( auto &ch : s ) {
        switch (ch)
        {
            case 'a':
                arr[i] = 1;
                break;
            case 'e':
                arr[i] = 1;
                break;
            case 'i':
                arr[i] = 1;
                break;
            case 'o':
                arr[i] = 1;
                break;
            case 'u':
                arr[i] = 1;
                break;
            default:
                arr[i] = 0;
                break;
        } i += 1;
    }
    
    int max_vowels = 0, sum_vowels = 0;
    for ( int i = 0; i < k; i++ )
        max_vowels += arr[i];
    sum_vowels = max_vowels;

    int j;
    for ( int i = 1; i < s.length()-k+1; i++ ) {
        j = i + k - 1;
        sum_vowels = sum_vowels - arr[i - 1] + arr[j];
        if ( max_vowels < sum_vowels )
            max_vowels = sum_vowels;
    }

    return max_vowels;
}